lib/pavilion/cmd_utils.py

```diff
@@ -118,3 +132,270 @@ def test_list_to_paths(pav_cfg, req_tests) -> List[Path]:
             test_paths.append(test_dir)
 
     return test_paths
+
+
+def get_test_configs(pav_cfg, host, test_files, tests, modes,
+                     overrides, logger=None, conditions=None, outfile=None):
+    """Translate a general set of pavilion test configs into the final,
+        resolved configurations. These objects will be organized in a
+        dictionary by scheduler, and have a scheduler object instantiated
+        and attached.
+
+    :param pav_cfg: The pavilion config.
+    :param str host: The host config to target these tests with
+    :param list(str) modes: The mode configs to use.
+    :param list(Path) test_files: Files containing a newline separated
+        list of tests.
+    :param list(str) tests: The tests to run.
+    :param list(str) overrides: Overrides to apply to the configurations.
+    :param Union[system_variables.SysVarDict,{}] sys_vars: The system
+        variables dict.
+    :returns: A dictionary (by scheduler type name) of lists of tuples
+        test configs and their variable managers.
+    """
+
+    if logger:
+        logger.debug("Finding Configs")
+
+    resolver = test_config.TestConfigResolver(pav_cfg)
+
+    tests = list(tests)
+    for file in test_files:
+        try:
+            with pathlib.PosixPath(file).open() as test_file:
+                for line in test_file.readlines():
+                    line = line.strip()
+                    if line and not line.startswith('#'):
+                        tests.append(line)
+        except (OSError, IOError) as err:
+            msg = "Could not read test file {}: {}".format(file, err)
+            if logger:
+                logger.error(msg)
+            raise commands.CommandError(msg)
+
+    try:
+        resolved_cfgs = resolver.load(
+            tests,
+            host,
+            modes,
+            overrides,
+            conditions=conditions,
+            output_file=outfile,
+        )
+    except TestConfigError as err:
+        raise commands.CommandError(err.args[0])
+
+    tests_by_scheduler = defaultdict(lambda: [])
+    for cfg, var_man in resolved_cfgs:
+        tests_by_scheduler[cfg['scheduler']].append((cfg, var_man))
+
+    return tests_by_scheduler
```

Comment:
Just return the list of tests. This scheduler organization thing was a mistake on my part.